src/timezones/Olson.jl

```diff
@@ -332,7 +341,10 @@ function resolve!(zone_name::AbstractString, zoneset::ZoneDict, ruleset::RuleDic
             end
         else
             if !haskey(ordered, rule_name)
-                ordered[rule_name] = order_rules(ruleset[rule_name]; max_year=max_year)
+                dates, ordered_rules, was_truncated = order_rules(ruleset[rule_name]; max_year=max_year)
+
+                ordered[rule_name] = dates, ordered_rules
+                truncated |= was_truncated
```

Comment:
I think this code is right. `resolve!` builds a single timezone and we want truncated to be `true` if any of the rules within that timezone are truncated. The `|=` should be enabling that behavior, right?